edb/edgeql/compiler/astutils.py

```diff
@@ -93,7 +93,9 @@ def is_ql_path(qlexpr: qlast.Expr) -> bool:
 
     start = qlexpr.steps[0]
 
-    return isinstance(start, (qlast.Source, qlast.ObjectRef, qlast.Ptr))
+    return isinstance(start, (qlast.ObjectRef, qlast.Ptr)) or (
+        isinstance(start, qlast.Anchor) and start.name == '__source__'
```

Comment:
Hm... Should other anchors be true here?

Reply:
Yes

Reply:
although this function is never used